cli/src/commands/mod.rs

```diff
@@ -2376,6 +2454,127 @@ Please use `jj new 'all:x|y'` instead of `jj new --allow-large-revsets x y`.",
+            [parent_id] => parent_id,
+            _ => return Err(user_error("Cannot run `jj next` on a merge commit")),
+        }
+    };
+    let target_expression = RevsetExpression::commit(start_id.clone())
+        .descendants_at(amount)
+        // Negate the current working copy.
+        .minus(&RevsetExpression::commit(current_wc_id.clone()));
+    let targets: Vec<Commit> = target_expression
+        .resolve(workspace_command.repo().as_ref())?
+        .evaluate(workspace_command.repo().as_ref())?
+        .iter()
+        .commits(workspace_command.repo().store())
+        .take(2)
+        .try_collect()?;
+    let target = match targets.as_slice() {
+        [target] => target,
+        [] => {
+            // TODO: How to handle the next onto new commit part?
+            //
+            // See below.
+            // If the parent is the last commit in the repository and we're not editing,
+            // just move the working-copy commit to the end. This is just a wrapped `jj
+            // new`.
+            return Err(user_error("No target_commit"));
+        }
+        _ => {
+            // TODO(#NNN) We currently cannot deal with multiple children, which result
+            // from branches. Prompt the user for resolution.
+            return Err(user_error("Ambiguous target commit"));
+        }
+    };
+    let target_short = short_commit_hash(target.id());
+    // If we're editing, the target must be rewritable.
+    if edit {
+        workspace_command.check_rewritable(target)?;
+    }
+    let mut tx = workspace_command.start_transaction("");
+    // We're editing, just move to the target commit.
+    if edit {
+        tx.set_description(&format!("next: {current_short} -> editing {target_short}"));
+        tx.edit(target)?;
+        tx.finish(ui)?;
+        return Ok(());
+    }
+    tx.set_description(&format!("next: {current_short} -> {target_short}"));
+    // Move the working-copy commit to the new parent.
+    tx.check_out(target)?;
+    Ok(())
+}
+
+fn cmd_prev(ui: &mut Ui, command: &CommandHelper, args: &PrevArgs) -> Result<(), CommandError> {
+    let mut workspace_command = command.workspace_helper(ui)?;
+    let edit = args.edit;
+    let amount = args.amount;
+    let current_wc_id = workspace_command
+        .get_wc_commit_id()
+        .ok_or_else(|| user_error("This command requires a working copy".to_string()))?;
+    let current_wc = workspace_command.repo().store().get_commit(current_wc_id)?;
+    let current_short = short_commit_hash(current_wc.id());
+    let start_id = if edit {
+        current_wc_id
+    } else {
+        match current_wc.parent_ids() {
+            [parent_id] => parent_id,
+            _ => return Err(user_error("cannot run `jj prev` on a branching parent")),
+        }
+    };
+    let target_revset = RevsetExpression::commit(start_id.clone())
+        .ancestors_at(amount)
+        .minus(&RevsetExpression::commit(current_wc_id.clone()));
```

Comment:
I don't think this can happen, unless `amount` is 0. Actually, I think we need to remove this line in order for that to work.